pkg/chunk/purger/purger_test.go

```diff
@@ -359,9 +359,6 @@ func TestPurger_Restarts(t *testing.T) {
 	// load in process delete requests by calling Run
 	require.NoError(t, services.StartAndAwaitRunning(context.Background(), newPurger))
 
-	// there must be 1 pending delete request
-	require.Equal(t, float64(1), testutil.ToFloat64(newPurger.metrics.pendingDeleteRequestsCount))
```

Comment:
This is a cause of flakyness. When this code is executed, the pending request may have already been picked up.